[test optimization] Add test session name `DD_TEST_SESSION_NAME` instructions (#27436)
DataDog/documentation: content/en/tests/setup/python.md

{{< tabs >}}
{{% tab "pytest" %}}

> To enable instrumentation of `pytest` tests, add the `--ddtrace` option when running `pytest`. Specify the test session name with the `DD_TEST_SESSION_NAME` environment variable, which identifies the group of tests about to run. Examples of this values are `unit-tests`, `integration-tests` or `smoke-tests`.

Reviewer: I feel that the way it is written may make it sound like there is a limited set of values that are accepted for the variable. Maybe switch to "would be" like in the list afterwards.

```suggestion
To enable instrumentation of `pytest` tests, add the `--ddtrace` option when running `pytest`. Specify the test session name with the `DD_TEST_SESSION_NAME` environment variable, which identifies the group of tests about to run. Examples of this value would be `unit-tests`, `integration-tests` or `smoke-tests`.
```

Reviewer: We also have to add similar text to the `unittest` tab.

    assert result == 25
```

> For additional configuration see [Configuration Settings][1].

Reviewer 2: ```suggestion
For additional configurations, see [Configuration Settings][1].
```

To enable instrumentation of `unittest` tests, run your tests by appending `ddtrace-run` to the beginning of your `unittest` command.

{{< code-block lang="shell" >}}
DD_TEST_SESSION_NAME=python-integration-tests ddtrace-run python -m unittest
{{< /code-block >}}

> Optionally specify the name of the service or library under test in the `DD_SERVICE` environment variable.

Reviewer: For pytest it has the "Additionally you may pass these environment variables:" block, but here we have this single sentence describing `DD_SERVICE` (but not `DD_ENV`). I think we should have the same instructions in both places.

Maybe we should have just a short description of `DD_SERVICE` and `DD_ENV` and an indication to see "Configuration Settings" below for more details.

Reviewer: This bit is still inconsistent between pytest and unittest :eyes: 

Besides the text, the `DD_TEST_SESSION_NAME` variable is specified in the unittest example but not in the pytest one.

Author: 👍 https://github.com/DataDog/documentation/pull/27436/commits/597ae83196cd453e4f25e0eb7e1fafae42e32d8a

assert True
{{< /code-block >}}

> For additional configuration see [Configuration Settings][1].

Reviewer 2: ```suggestion
For additional configurations, see [Configuration Settings][1].
```

    api.TestSession.finish()
```

> For additional configuration see [Configuration Settings][2].

Reviewer 2: ```suggestion
For additional configurations, see [Configuration Settings][2].
```

The following is a list of the most important configuration settings that can be used with the tracer, either in code or using environment variables:

`DD_TEST_SESSION_NAME`

> : Use it to identify a group of tests, such as `integration-tests`, `unit-tests` or `smoke-tests`.<br/>

Reviewer 2: ```suggestion
: Identifies a group of tests, such as `integration-tests`, `unit-tests` or `smoke-tests`.<br/>
```

{{% ci-git-metadata %}}

## Best practices

### Test session name `DD_TEST_SESSION_NAME`

> Use `DD_TEST_SESSION_NAME` to define the test session name for your tests (`test_session.name` tag). Use this to identify a group of tests. Examples of values for this tag would be:

Reviewer 2: Same as above comment

Author: https://github.com/DataDog/documentation/pull/27436/files#r1946692292

- `unit-tests`
- `integration-tests`
- `smoke-tests`
- `flaky-tests`
- `ui-tests`
- `backend-tests`

> If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of:

Reviewer 2: ```suggestion
If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of the:
```

- CI job name
- Command used to run the tests (such as `pytest --ddtrace`)

> The test session name should be unique within a repository to help you distinguish different groups of tests.

Reviewer 2: ```suggestion
The test session name needs to be unique within a repository to help you distinguish different groups of tests.
```

#### When to use `DD_TEST_SESSION_NAME`

If your tests are run with commands that include a dynamic string, such as:

- `pytest --temp-dir=/var/folders/t1/rs2htfh55mz9px2j4prmpg_c0000gq/T`

> The default value for the test session name will be unstable. It is recommended to use `DD_TEST_SESSION_NAME` in this case.

Reviewer 2: Same question as above

Author: replied above

{{% ci-git-metadata %}}

## Best practices

### Test session name `DD_TEST_SESSION_NAME`

Use `DD_TEST_SESSION_NAME` to define the test session name for your tests (`test_session.name` tag). Use this to identify a group of tests. Examples of values for this tag would be:

- `unit-tests`
- `integration-tests`
- `smoke-tests`
- `flaky-tests`
- `ui-tests`
- `backend-tests`

If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of:

- CI job name

> - Command used to run the tests (such as `yarn test`)

Reviewer: ```suggestion
- Command used to run the tests (such as `pytest --ddtrace`)
```
